assets/scripts/admin/surveysettings.js

```diff
@@ -152,6 +152,38 @@ function guidGenerator() {
     return (S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4());
 }
 
+/**
+ * Validate settings form depending on form name
+ */
+function validateSettingsForm($form) {
+    switch ($form.attr('id')) {
+        case 'publication':
+            return validateEndDate();
+        default:
+            return true;
+    }
+}
+
+/**
+ * Validates that the expiration date is not lower than the start date
+ */
+function validateEndDate() {
```

Comment:
Do we need to duplicate this logic in javascript? Isn't the model validation enough?

Reply:
Not on the question editor.
As this screen covers so many models, if one model throughs an error, the already inputed data may be lost.
So, we validate as much as possible on the front end to avoid that.

Reply:
OK. Well, can you pass in $form to this function too?

Reply:
Hm, maybe better would be the IDs of the datetime pickers. Now they are hardcoded, which makes the function not usable in other contexts.

Reply:
My opinion: we must check both ...

1. disallow it in model (**must be done**)
2. return an error in model
3. Get error in form and show it in a clean HTML
5. Unsure we don't update sent value (all we can)
6. Add the JS/HTML5 validation to disallow submit
7. get error in remote control and show it as clean HTML

Reply:
> Or better still, send in the $startdateTimepicker and $expiresTimepicker.

Ready for re-review